core/engine/src/builtins/date/utils.rs

```diff
@@ -1,202 +1,730 @@
+    // 5. Let milli be 𝔽(! ToIntegerOrInfinity(ms)).
+    let milli = ms.abs().floor().copysign(ms);
 
-    h_ms.checked_add(m_ms)?.checked_add(s_ms)?.checked_add(ms)
+    // 6. Return ((h × msPerHour + m × msPerMinute) + s × msPerSecond) + milli.
+    ((h * MS_PER_HOUR + m * MS_PER_MINUTE) + s * MS_PER_SECOND) + milli
 }
 
-/// Abstract operation [`MakeDay`][spec].
+/// Abstract operation `MakeDay ( year, month, date )`
+///
+/// More info:
+/// - [ECMAScript reference][spec]
 ///
-/// [spec]: https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-makeday
-pub(crate) fn make_day(mut year: i64, mut month: i64, date: i64) -> Option<i64> {
-    // 1. If year is not finite or month is not finite or date is not finite, return NaN.
+/// [spec]: https://tc39.es/ecma262/#sec-makeday
+pub(super) fn make_day(year: f64, month: f64, date: f64) -> f64 {
+    // 1. If year is not finite, month is not finite, or date is not finite, return NaN.
+    if !year.is_finite() || !month.is_finite() || !date.is_finite() {
+        return f64::NAN;
+    }
+
     // 2. Let y be 𝔽(! ToIntegerOrInfinity(year)).
+    let y = year.abs().floor().copysign(year);
+
     // 3. Let m be 𝔽(! ToIntegerOrInfinity(month)).
-    // 4. Let dt be 𝔽(! ToIntegerOrInfinity(date)).
-    if !(MIN_YEAR..=MAX_YEAR).contains(&year) || !(MIN_MONTH..=MAX_MONTH).contains(&month) {
-        return None;
-    }
+    let m = month.abs().floor().copysign(month);
 
-    // At this point, we've already asserted that year and month are much less than its theoretical
-    // maximum and minimum values (i64::MAX/MIN), so we don't need to do checked operations.
+    // 4. Let dt be 𝔽(! ToIntegerOrInfinity(date)).
+    let dt = date.abs().floor().copysign(date);
 
     // 5. Let ym be y + 𝔽(floor(ℝ(m) / 12)).
+    let ym = y + (m / 12.0).floor();
+
     // 6. If ym is not finite, return NaN.
-    year += month / 12;
-    // 7. Let mn be 𝔽(ℝ(m) modulo 12).
-    month %= 12;
-    if month < 0 {
-        month += 12;
-        year -= 1;
+    if !ym.is_finite() {
+        return f64::NAN;
     }
 
-    // 8. Find a finite time value t such that YearFromTime(t) is ym and MonthFromTime(t) is mn and DateFromTime(t) is
-    // 1𝔽; but if this is not possible (because some argument is out of range), return NaN.
-    let month = usize::try_from(month).expect("month must be between 0 and 11 at this point");
-
-    let mut day = day_from_year(year);
-
-    // Consider leap years when calculating the cumulative days added to the year from the input month
-    if (year % 4 != 0) || (year % 100 == 0 && year % 400 != 0) {
-        day += [0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334][month];
-    } else {
-        day += [0, 31, 60, 91, 121, 152, 182, 213, 244, 274, 305, 335][month];
-    }
+    // 7. Let mn be 𝔽(ℝ(m) modulo 12).
+    let mn = modulo(m, 12.0) as u8;
+
+    // 8. Find a finite time value t such that YearFromTime(t) is ym, MonthFromTime(t) is mn,
```

Comment:
I think there's an outstanding issue on `MakeDay` here. Idk if it would be worth tracking it. 

For [reference](https://github.com/tc39/ecma262/issues/1087)